### PR TITLE
refactor: change annotations to use built-in generic types

### PR DESCRIPTION
**Pull request by Carlos-Muniz**
Changes annotations from `typing` to use built-in generic types from `__future__.annotations`

Closes https://github.com/openedx/wg-developer-experience/issues/130

### CODE REVIEW

**Comment by Carlos-Muniz** [comment]
@kdmccormick @regisb This is ready for review

**Review by regisb — CHANGES_REQUESTED** [review]
Looking good, thanks! Just a few minor comments.

File: tests/test_plugins_v0.py [hunk 1]
```diff
@@ -197,9 +198,9 @@ def test_dict_plugin(self) -> None:
             {"name": "myplugin", "config": {"set": {"KEY": "value"}}, "version": "0.1"}
         )
         plugins.load("myplugin")
-        overriden_items: t.List[
-            t.Tuple[str, t.Any]
-        ] = hooks.Filters.CONFIG_OVERRIDES.apply([])
+        overriden_items: list[tuple[str, t.Any]] = hooks.Filters.CONFIG_OVERRIDES.apply(
```
[the remark this answers is not in the document]
Comment (Carlos-Muniz) [reply tests/test_plugins_v0.py:201]: I'm not sure I understand. Do you want me to remove  `: list[tuple[str, t.Any]]` from line 201?

[the remark this answers is not in the document]
Comment (Carlos-Muniz) [reply tests/test_plugins_v0.py]: Removed `: list[tuple[str, t.Any]]`

File: tests/commands/test_compose.py [hunk 2]
```diff
@@ -78,11 +79,11 @@ def test_compose_local_tmp_generation(self, _mock_stdout: StringIO) -> None:
         }
         self.assertEqual(actual_services, expected_services)
 
-        compose_jobs_file: t.Dict[
+        compose_jobs_file: dict[
```
[the remark this answers is not in the document]
Comment (Carlos-Muniz) [reply tests/commands/test_compose.py:82]: The mypy tests only pass if the type annotations are removed for both `compose_jobs_file` and `actual_jobs_services`.

[the remark this answers is not in the document]
Comment (regisb) [reply tests/commands/test_compose.py:82]: Right, let's get rid of them.

[the remark this answers is not in the document]
Comment (Carlos-Muniz) [reply tests/commands/test_compose.py]: the type annotations have been removed for both compose_jobs_file and actual_jobs_services.

**Review by regisb — APPROVED** [review]
Thanks for your contribution!

**Comment by regisb** [comment]
I realized only now that your PR was on the nightly branch. I'll backport the commit to the master branch.

**Pull request merged**